### ISSUE TITLE
False positive UnusedPrivateElements call on static function return

### ISSUE THREAD

**Issue by liepumartins** [slevomat/coding-standard#905]
Following example, methods are just to mimic the behaviour.
```php
<?php

class Date
{
    private $datetime;

    public static function getAvailable(): DateTime
    {
        $specific = self::getSpecific();
        if ($specific) {
            return $specific->dateTime();
        }
        return new DateTime();
    }

    private static function getSpecific(): ?Date
    {
        return new Date();
    }

    private function dateTime(): DateTime
    {
        return DateTime::createFromFormat('Y-m-d H:i:s', $this->datetime);
    }
}
```

Sniff incorrectly, that `dateTime()` is unused:
```
[Code] Unused private elements:
  Date.php:21: Class Date contains unused private method dateTime().
```

**Comment by kukulich** [comment]
@liepumartins Can you try `self` instead of `Date` after `new`?

**Comment by liepumartins** [comment]
Yes, same result.

```php
private static function getSpecific(): ?Date
{
    return new self();
}
```

Also same result changing `self::getSpecific()` to `Date::getSpecific()`
```php
    public static function getAvailable(): DateTime
    {
        $specific = Date::getSpecific();
        if ($specific) {
            return $specific->dateTime();
        }
        return new DateTime();
    }
```

**Comment by kukulich** [comment]
It's too difficult to solve it in PHPCS.
